hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -184,8 +193,25 @@ public void onConfigChange(ConfigChangeEvent tableEvent) throws MetaException {
           TimeUnit.SECONDS);
       MetastoreConf.setTimeVar(getConf(), MetastoreConf.ConfVars.EVENT_DB_LISTENER_TTL, time,
           TimeUnit.SECONDS);
-      cleaner.setTimeToLive(MetastoreConf.getTimeVar(getConf(),
-          MetastoreConf.ConfVars.EVENT_DB_LISTENER_TTL, TimeUnit.SECONDS));
+      boolean isReplEnabled = MetastoreConf.getBoolVar(getConf(), ConfVars.REPLCMENABLED);
+      if(!isReplEnabled){
+        cleaner.setTimeToLive(MetastoreConf.getTimeVar(getConf(), ConfVars.EVENT_DB_LISTENER_TTL,
+                TimeUnit.SECONDS));
+      }
+    } else if (key.equals(ConfVars.REPL_EVENT_DB_LISTENER_TTL.toString()) ||
+            key.equals(ConfVars.REPL_EVENT_DB_LISTENER_TTL.getHiveName())) {
+      // This weirdness of setting it in our conf and then reading back does two things.
+      // One, it handles the conversion of the TimeUnit.  Two, it keeps the value around for
+      // later in case we need it again.
+      long time = MetastoreConf.convertTimeStr(tableEvent.getNewValue(), TimeUnit.SECONDS,
+              TimeUnit.SECONDS);
+      MetastoreConf.setTimeVar(getConf(), MetastoreConf.ConfVars.REPL_EVENT_DB_LISTENER_TTL, time,
```

Comment:
This line updates the value of the configuration. Line 212 sets the new value in the cleaner's parameters.

Reply:
why do you need to update value of config. You can do a getTimeVar in the unit you want

Reply:
Updates to the existing config( EVENT_DB_LISTENER_TTL) was dealt with in the same manner. Not sure why it was required in the first place.

Reply:
// This weirdness of setting it in our conf and then reading back does two things.
      // One, it handles the conversion of the TimeUnit.  Two, it keeps the value around for
      // later in case we need it again.

See if this is used later or its just a one time thing. Then you can go ahead without setting in conf I think.